### ISSUE TITLE
[Enhancement][a11y] Улучшить доступность `ChipsInput` и `ChipsSelect`

### ISSUE THREAD

**Issue by eugpoloz** [VKCOM/VKUI#4143]
Сейчас доступность `ChipsInput` и `ChipsSelect` основана на [application role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/application_role). Эта роль — не самая безопасная:

> [Accessibility concerns](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/application_role#accessibility_concerns)
Improperly using the application role can unintentionally take away access from information on a web page, so be very mindful of using it. Think hard on if you actually need it and cannot just use a set of other known widgets to accomplish the same task.

**Как можно сделать лучше?**
- переехать на [combobox role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role) и [grid role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/grid_role) (так, например, реализованы [чипы в Material](https://material.angular.io/components/chips/overview))
- поддержать навигацию стрелочками внутри чипов 

---

# <a href="#user-content-logic" id="logic">🔗</a> UPD. Логика работы

> Добавленные в пункты будут обозваны чипами, чтобы не было путаницы с пунктами из списка.

## Кейс 1. Ничего ещё не добавили.

Мы заходим в `ChipsInput`/`ChipsSelect` и попадаем на поле ввода (см. _Примечание 1_). После набора текста, нажатие **Enter** создаст чип (см. _Примечание 2_).

## Кейс 2. Есть добавленные чипы

Если мы заходим в `ChipsInput`/`ChipsSelect` через **Tab**, то сфокусируемся на чип, если этих чипов несколько, то мы можем навигироваться между ними стрелками. Ещё один **Tab** сфокусирует на поле ввода, что сразу вызовет список с автозаполнением ну и дальше всё как описано в _Кейсе 1_. При **Shift + Tab**, всё тоже самое, но в первый раз мы попадём на поле вводе, ещё один **Shift + Tab** сфокусирует на чип (см. _Примечание 3_). Если мы заходим в `ChipsInput`/`ChipsSelect` мышкой, то мы можем сфокусироваться либо на каком-то из чипов, либо на поле ввода.

--- 

####  Примечание 1

В `ChipsSelect` фокус на поле ввода сразу разворачивает список с автозаполнением, где пункт из этого выпадающего списка мы можем выбрать либо стрелками **ArrowUp**/**ArrowDown** и кнопкой **Enter**, либо мышкой. Поле ввода по умолчанию фильтрует пункты в списке. Например, нам нужен пункт из списка под названием **Linkin Park**, мы вводим часть названия и список отфильтруется по совпадениям. Дальше теми же стрелками или мышкой выбираем пункт в отфильтрованном списке. Выбранный пункт будет отмечен галочкой и атрибутом `aria-selected` со значением `true`.

#### Примечание 2

В `ChipsSelect` эту возможность надо включить через параметр `creatable`, чтобы можно было добавлять пункт, которого нет в списке. Рассмотрим на примере создания хэштегов. Пользователь начинает набирать хэштег, в списке автозаполнения фильтруются и подсказываеются известные хэштеги. Когда пользователь нажимает **Enter**, создаётся хэштег из набранного им текста. Помимо **Enter**, есть возможность отрисовывать в выпадающем списке единственный пункт с произвольным текстом, например, "добавить новый хэштег", поведение на нажатие (либо стрелками, либо мышкой) такое же, что при **Enter** в поле ввода.

#### Примечание 3

Подробности про фокус на чип. При **Tab** или **Shift + Tab** первым фокус получает тот чип, который в последний раз имел фокус.

**Comment by mendrew** [comment]
@inomdzhon тут ещё прилетел прекрасный PR на добавление кнопки очистки в ChipsSelect. Посмотришь в рамках своей работы? Не конфликтует ли. Скорее всего конфликтует, и тут дилема, просить ли исправлять некоторые вещи уже сейчас или подождать результатов твоей работы.
https://github.com/VKCOM/VKUI/pull/6189

**Comment by inomdzhon** [comment]
> @inomdzhon тут ещё прилетел прекрасный PR на добавление кнопки очистки в ChipsSelect. Посмотришь в рамках своей работы? Не конфликтует ли. Скорее всего конфликтует, и тут дилема, просить ли исправлять некоторые вещи уже сейчас или подождать результатов твоей работы. #6189

Ага, видел. Я бы дал в приоритет PR'у от контрибьтера, но придётся моё первым пустить, т.к. мне надо для перерлиза катиться 😥

**Comment by inomdzhon** [comment]
Коллега по визуальной доступности обнаружил, что [NVDA](https://www.nvaccess.org/) не может найти `ChipsInput` и `ChipsSelect` при навигации мышью.

Задачу переоткрываю, т.к. #6206 не решил проблему.

**Comment by vkcom-publisher** [comment]
✅ <a href="https://github.com/VKCOM/VKUI/releases/tag/v6.0.0" target="_blank">v6.0.0</a> 🎉
